extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -838,19 +839,29 @@ export class AmpStoryEmbeddedComponent {
       user().error(TAG, 'The tooltip icon url is invalid');
       return;
     }
-    const iconSrc = iconUrl ? parseUrlDeprecated(iconUrl).href :
-      embedConfig.componentIcon;
 
-    const existingTooltipIcon =
+    const tooltipCustomIcon =
       this.tooltip_.querySelector('.i-amphtml-story-tooltip-icon');
 
-    if (existingTooltipIcon.firstElementChild) {
-      addAttributesToElement(existingTooltipIcon.firstElementChild,
-          dict({'src': iconSrc}));
+    // Publisher specified a valid icon url.
+    if (iconUrl) {
+      this.resources_.mutateElement(devAssert(tooltipCustomIcon), () => {
+        setImportantStyles(devAssert(tooltipCustomIcon),
+            {'background-image': `url(${parseUrlDeprecated(iconUrl).href})`});
+      });
+      return;
+    }
+
+    // No icon src specified by publisher and no default icon in config.
+    if (embedConfig.componentIcon === BLANK_ICON_SRC) {
+      tooltipCustomIcon.classList.toggle('i-amphtml-hidden', true);
+      return;
     }
 
-    existingTooltipIcon.classList.toggle('i-amphtml-hidden',
-        iconSrc == DEFAULT_ICON_SRC);
+    // No icon src specified by publisher. Use default icon found in the config.
+    this.resources_.mutateElement(devAssert(tooltipCustomIcon), () => {
+      tooltipCustomIcon.classList.add(embedConfig.componentIcon);
```

Comment:
What do you think of changing this property name for when it's a class? It's a bit weird that it could be either a class or a src, and that we're relying on the `if` order to know what it is.